brave-core/src/main/java/com/github/kristofa/brave/HttpSpanCollector.java

```diff
@@ -0,0 +1,234 @@
+    return new HttpSpanCollector(baseUrl, Config.builder().build(), metrics);
+  }
+
+  /**
+   * @param baseUrl URL of the zipkin query server instance. Like: http://localhost:9411/
+   * @param config controls flush interval and timeouts
+   * @param metrics Gets notified when spans are accepted or dropped. If you are not interested in
+   * these events you can use {@linkplain EmptySpanCollectorMetricsHandler}
+   */
+  public static HttpSpanCollector create(String baseUrl, Config config,
+      SpanCollectorMetricsHandler metrics) {
+    return new HttpSpanCollector(baseUrl, config, metrics);
+  }
+
+  // Visible for testing. Ex when tests need to explicitly control flushing, set interval to 0.
+  HttpSpanCollector(String baseUrl, Config config,
+      SpanCollectorMetricsHandler metrics) {
+    this.url = baseUrl + (baseUrl.endsWith("/") ? "" : "/") + "api/v1/spans";
+    this.metrics = metrics;
+    this.config = config;
+    this.flusher = config.flushInterval() > 0 ? new Flusher(this, config.flushInterval()) : null;
+  }
+
+  /**
+   * Queues the span for collection, or drops it if the queue is full.
+   *
+   * @param span Span, should not be <code>null</code>.
+   */
+  @Override
+  public void collect(Span span) {
+    metrics.incrementAcceptedSpans(1);
+    if (!pending.offer(span)) {
+      metrics.incrementDroppedSpans(1);
+    }
+  }
+
+  /**
+   * Calling this will flush any pending spans to the http transport on the current thread.
+   */
+  @Override
+  public void flush() {
+    if (pending.isEmpty()) return;
+    Queue<Span> drained = new ArrayDeque<>(pending.size());
+    pending.drainTo(drained);
+    if (drained.isEmpty()) return;
+
+    // Thrift-encode the spans for transport
+    int spanCount = drained.size();
+    BufferTransport thrifts = new BufferTransport();
+    try {
+      TBinaryProtocol oprot = new TBinaryProtocol(thrifts);
+      oprot.writeListBegin(new TList(TType.STRUCT, drained.size()));
+      while (!drained.isEmpty()) {
+        drained.remove().write(oprot);
+      }
+      oprot.writeListEnd();
+    } catch (TException e) {
+      metrics.incrementDroppedSpans(spanCount);
+      return;
+    }
+
+    // Send the thrifts to the zipkin endpoint
+    try {
+      postSpans(thrifts.buffer);
+    } catch (IOException e) {
+      metrics.incrementDroppedSpans(spanCount);
+      return;
+    }
+  }
+
+  /** Calls flush on a fixed interval */
+  static final class Flusher implements Runnable {
+    final Flushable flushable;
+    final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+
+    Flusher(Flushable flushable, int flushInterval) {
+      this.flushable = flushable;
+      this.scheduler.scheduleWithFixedDelay(this, 0, flushInterval, SECONDS);
+    }
+
+    @Override
+    public void run() {
+      try {
+        flushable.flush();
+      } catch (IOException ignored) {
+      }
+    }
+  }
+
+  void postSpans(Buffer thrifts) throws IOException {
```

Comment:
future extension point, but hidden for now. I'll almost certainly make an okhttp implementation